src/backend/dx12/src/format.rs

```diff
@@ -0,0 +1,1123 @@
+//! Format related queries for the backend.
+
+use hal::format::{BufferFeature, ImageFeature, Properties, NUM_FORMATS};
+
+///
+pub fn query_properties() -> [Properties; NUM_FORMATS] {
+    // TODO
+    let properties = [
+        // Undefined
+        Properties {
```

Comment:
perhaps, we could use `default()` more aggressively, e.g.
```rust
Properties {
    optimal_tiling: ImageFeature::SAMPLED | ImageFeature::BLIT_SRC | ImageFeature::SAMPLED_LINEAR , 
    .. Properties::default()
}
```

Although, on a second thought, this would be harmful when we actually fill all of those up, at which point it will be clearer to see if some helper constructors would work best, so no point doing anything now.